### PR TITLE
London-Class7-JavaScript-Core-2-Coursework-Week1-AhmadJanah

### PR DESCRIPTION
**Pull request by AhmadJanah**
# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

### CODE REVIEW

File: mandatory/8-cheap-diner.js [hunk 1]
```diff
@@ -31,6 +31,35 @@ Should give the answer "Nothing :("
...
   // Write your code here
+  if (mealArray.length == 0)
+    return "Nothing :(";
+  else if (mealArray.length == 1)
+    return mealArray[0].name;
+
+  // let newArr = mealArray.map(element => {
+  //   return element.price;
+  // }).sort(function(a, b){return a - b})[1];
+  let cheap1 = 1000;
```
Comment (iTrip23) [inline mandatory/8-cheap-diner.js:42]: I like your solution to the problem, nice approach.
Happy coding!

File: mandatory/1-writers.js [hunk 2]
```diff
@@ -69,10 +76,33 @@ Exercise 2:
...
+//   writer.alive === false && writer.age> 40 && writer.age<50
+// })
+// console.log(`Writer ${writer.firstName} ${writer.lastName} died at ${writer.age} years old.`);
+// })
+
+writers.filter(writer => {
+  return writer.alive === false && writer.age> 40 && writer.age<50
+}).map(writer => {
+  console.log(`Writer ${writer.firstName} ${writer.lastName} died at ${writer.age} years old.`);
+})
```
Comment (syedur-r) [inline mandatory/1-writers.js:94]: Great work on adding multiple conditions to filter out the age of writers. 

Although the outputs are correct, you may want to have another look at the instructions which says to check age range between 40 and 49 (including 40). Your age condition excludes 40 year old writers, and only checks 41 to 49 year olds. How would you include 40 in your condition?

File: mandatory/1-writers.js [hunk 3]
```diff
@@ -69,10 +76,33 @@ Exercise 2:
...
   Only `console.log()` out alive writers who are in their 40s (meaning between 40 and 49):
 
   "Hi, my name is {firstName} {lastName}. I am {age} years old."
 */
+
+writers.filter(writer => {
+  return writer.alive === true && writer.age> 40 && writer.age<50
+}).map(writer => {
+  console.log(`Hi, my name is ${writer.firstName} ${writer.lastName}. I am ${writer.age} years old.`);
+})
```
Comment (syedur-r) [inline mandatory/1-writers.js:108]: The same applies here, you need to include 40 in your condition. How would you do that?

File: mandatory/1-writers.js (hunk 2 above)
Comment (AhmadJanah) [reply mandatory/1-writers.js:94]: Hi Syedur,

Thanks for your review. I am happy that this is the only issue. we can amend it to be: writer.age>= 40
It is really appreciated.

Regrds

**Comment by syedur-r** [comment]
Excellent! Keep up the great work!

File: mandatory/4-people-I-know.js [hunk 4]
```diff
@@ -388,6 +388,14 @@ First, I want you to find all of my friends who are 35 or older.
 
 let thirtyFiveOrOlder = [];
```
Comment (11ma) [inline mandatory/4-people-I-know.js:390]: Nice use of the higher-order functions.

File: mandatory/4-people-I-know.js [hunk 5]
```diff
@@ -411,6 +427,12 @@ This time, I only want the full names of the people are who friends with her.
 
 let friendsWithStacie = [];
```
Comment (11ma) [inline mandatory/4-people-I-know.js:429]: Good work over here 👏

**Pull request closed without merging**